hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -54,6 +60,18 @@ public class AbfsInputStream extends FSInputStream {
   //                                                      of valid bytes in buffer)
   private boolean closed = false;
 
+  //  Optimisations modify the pointer fields.
+  //  For better resilience the following fields are used to save the
+  //  existing state before optimisation flows.
+  private int limitBkp;
+  private int bCursorBkp;
+  private long fCursorBkp;
+  private long fCursorAfterLastReadBkp;
+
+  private boolean firstRead = true;
+  private boolean isReadSmallFilesCompletelyEnabled = true;
+  private boolean isOptimizeFooterReadEnabled = true;
```

Comment:
Lets have them final?

Or should we have config with default true?

Reply:
Yes, final makes sense.
They were configs only in the original commit. Expectation here is to confine all production code changes to a single file.